exercises/pascals-triangle/src/example.c

```diff
@@ -0,0 +1,26 @@
+#include "pascals_triangle.h"
+
+long long unsigned int factorial(int n)
+{
+   return (n == 0) ? 1 : n * factorial(n - 1);
+}
+
+long long unsigned int combination(int n, int r)
+{
+   return (r > n) ? 0 : factorial(n) / (factorial(n - r) * factorial(r));
+}
+
+int pascals_triangle(int count, int input[][((count < 0) ? 1 : count)])
+{
+   if (count < 0) {
+      memset(input, -1, sizeof(int) * count * count);
+      return -1;
+   }
+   memset(input, 0, sizeof(int) * count * count);
```

Comment:
we cannot give a warning , because it must handle anything quietly and canonical-data wants to return a 0 row array anyway , (where C89 we cannot assign int array[0][0]) , so i used a single row